### PR TITLE
Xrd modification

### PR DESCRIPTION
**Pull request by RubelMozumder**
1. Dissolve the scattering vector into perpendicular and parallel components (extension).
2. Rename the plot name of `two_theta_plot` compatible with the ELNXRDDiffraction schema.

### CODE REVIEW

**Review by ka-sarthak — COMMENTED** [review]
The quantity labels look good to me. I am not sure about the descriptions of the scatter vector components. If you took it from a reputable source, then it's okay, otherwise please check it with an expert in Area B, perhaps Sandor.

**Comment by RubelMozumder** [comment]
> The quantity labels look good to me. I am not sure about the descriptions of the scatter vector components. If you took it from a reputable source, then it's okay, otherwise please check it with an expert in Area B, perhaps Sandor.

`q` is a scattering vector; therefore, it has components along the dimension of the phase space or vector space. Here, the fields explain, the perpendicular component and parallel components of the scattering vector, the parallel component is important in the Reciprocal Space Mapping technique in the X-ray diffraction domain.

File: contributed_definitions/nyaml/NXxrd_pan.yaml [hunk 1]
```diff
@@ -146,42 +146,75 @@ NXxrd_pan(NXxrd):
...
+        doc: |
+          The norm of scattering vector q.
+    q_data(NXdata):
       exists: optional
       doc: |
         Desired plot would be q_vector vs intensity.
       q(NX_FLOAT):
         doc: |
-          Axis scale representing wavevector for scatter energy.
+          Scattering vector link to the 'q_norm' of experiment_result.
```
Comment (domna) [inline contributed_definitions/nyaml/NXxrd_pan.yaml:200]: It's a bit unclear what you mean here.

Generally, links cannot be enforced in Nexus, so you should also add a description for this value and then explain that it should be linked to `q_norm`. You can also state that these two are the same concept (unfortunately only in text form yet).

File: contributed_definitions/nyaml/NXxrd_pan.yaml [hunk 2]
```diff
@@ -146,42 +146,75 @@ NXxrd_pan(NXxrd):
...
       intensity(NX_FLOAT):
         doc: |
-          Number of scattered electrons per unit time.
+          Number of scattered electrons per unit time at each q value.
+          
+          Link to the 'intensity' of experiment_result.
```
Comment (domna) [inline contributed_definitions/nyaml/NXxrd_pan.yaml:205]: I would reformulate to something like `This should be linked to 'ENTRY/experiment_result/intensity'`

**Review by domna — COMMENTED** [review]
I cannot say anything about the scientific reasonability of these fields. From my very limited knowledge of XRD this reads fine though.

I left some comments on how to deal with links. [Here](https://fairmat-nfdi.github.io/nexus_definitions/classes/contributed_definitions/NXmpes.html#nxmpes-entry-sample-bias-env-voltmeter-group) is an example how we formulate this in NXmpes.

File: contributed_definitions/nyaml/NXxrd_pan.yaml (hunk 1 above)
Comment (RubelMozumder) [reply contributed_definitions/nyaml/NXxrd_pan.yaml]: Noted!

File: contributed_definitions/nyaml/NXxrd_pan.yaml (hunk 2 above)
Comment (RubelMozumder) [reply contributed_definitions/nyaml/NXxrd_pan.yaml]: Noted!

File: contributed_definitions/nyaml/NXxrd_pan.yaml [hunk 3]
```diff
@@ -136,52 +136,94 @@ NXxrd_pan(NXxrd):
         unit: NX_TIME
         doc: |
           Total time of count.
-    two_theta_plot(NXdata):
+    experiment_result(NXdata):
       doc: |
-        Desired plot would be two_theta vs intensity.
+        All the experiment results data such as scattering angle (2theta),
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXxrd_pan.yaml:141]:
```suggestion
        All experiment results data such as scattering angle (2theta),
```

**Review by domna — COMMENTED** [review]
Some more wording suggestions.

Generally, this addresses these points throughout:

- Avoid general phrases as `This concept represents ...`. It is clear that this is explaining the concept at it makes getting the relevant information quickly harder.
- Avoid referring to something as `... this is a link to...`, because we cannot enforce linking on the appdef level. We can just _suggest_ the user to link this data when writing the file. I would therefore suggest to word it as `... this should be linked to...` (for same-as concepts) or `... this might be linked to ...` (for similar concepts which are the same in some cases but not in others).

You might also want to add some literature which explains the scientific background of, e.g., q-vectors. However, I do believe this is not strictly necessary as we always assume that users of the appdef are also experts in the fields (but it cannot hurt if non-experts can also gain some understanding).

File: contributed_definitions/nyaml/NXxrd_pan.yaml [hunk 4]
```diff
@@ -136,52 +136,94 @@ NXxrd_pan(NXxrd):
...
+      q_norm(NX_FLOAT):
+        exists: optional
+        unit: NX_ANY
+        doc: |
+          The concept represents the norm value of the scattering vector, q.
+    q_data(NXdata):
       exists: optional
       doc: |
-        Desired plot would be q_vector vs intensity.
+        The desired plot would be for scattering vectors.
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXxrd_pan.yaml:198]: This does not make sense? I guess it should be `The desired view for scattering vectors`.

**Review by lukaspie — CHANGES_REQUESTED** [review]
I second what @domna commented, some more suggestions

File: contributed_definitions/nyaml/NXxrd_pan.yaml [hunk 5]
```diff
@@ -136,52 +136,94 @@ NXxrd_pan(NXxrd):
...
       q(NX_FLOAT):
+        exists: optional
         doc: |
-          Axis scale representing wavevector for scatter energy.
+          This concept corresponds to the norm value of the scattering vector(q).
```
[the remark this answers is not in the document]
Comment (RubelMozumder) [reply contributed_definitions/nyaml/NXxrd_pan.yaml:205]: Added.

**Review by domna — APPROVED** [review]
Looks good from my side now. Thanks

File: contributed_definitions/nyaml/NXxrd_pan.yaml [hunk 6]
```diff
@@ -136,52 +136,97 @@ NXxrd_pan(NXxrd):
...
+    q_data(NXdata):
       exists: optional
       doc: |
-        Desired plot would be q_vector vs intensity.
+        The desired view for scattering vectors.
       q(NX_FLOAT):
+        exists: optional
         doc: |
-          Axis scale representing wavevector for scatter energy.
+          This concept corresponds to the norm value of the scattering vector(q).
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXxrd_pan.yaml:205]:
```suggestion
          Normalized value of the scattering vector(q).
```

**Review by lukaspie — APPROVED** [review]
Found one more instance where things can be shortened. Other than that LGTM.

**Pull request merged**